vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -924,6 +924,11 @@ func resourceVSphereComputeClusterApplyClusterConfiguration(
 
 	log.Printf("[DEBUG] %s: Applying cluster configuration", resourceVSphereComputeClusterIDString(d))
 
+	// handle VSAN first to avoid race condition
```

Comment:
@SlimYang I've opted to refactor the code to just always process VSAN reconfigures first, is that safe under all scenarios? Personally in provider development I tend to lean towards calling code redundantly if it keeps the provider codebase clearer of conditional logic (try and make the provider codebase as idempotent as possible, unless excessive API use becomes a performance problem)

Reply:
I think it will cause issues when we are reconfiguring the vSAN and HA together. Specifically, we need to enable vSAN before HA enable (see the second note from https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vsan-planning.doc/GUID-D68890D8-841A-4BD1-ACA1-DA3D25B6A37A.html). Otherwise we will hit errors and the original code is to handle this scenario.